frontend/public/components/modals/configure-cluster-upstream-modal.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+
+import { ClusterVersionModel } from '../../models';
+import { ClusterVersionKind, k8sPatch } from '../../module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '../factory/modal';
+import { ExternalLink, HandlePromiseProps, withHandlePromise } from '../utils';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { RadioInput } from '../radio';
+import {
+  CLUSTER_VERSION_DEFAULT_UPSTREAM_SERVER_URL,
+  CLUSTER_VERSION_CUSTOM_UPSTREAM_SERVER_URL_PLACEHOLDER,
+} from '@console/shared/src/constants';
+import { TextInput } from '@patternfly/react-core';
+
+export const ConfigureClusterUpstreamModal = withHandlePromise(
+  (props: ConfigureClusterUpstreamModalProps) => {
+    const { cv, handlePromise, close } = props;
+    const currentUpstream = cv?.spec?.upstream;
+
+    const [selectedRadio, setSelectedRadio] = React.useState(
+      currentUpstream ? 'custom' : 'default',
+    );
+    const [customUrl, setCustomUrl] = React.useState(currentUpstream);
+    const [invalidCustomUrl, setInvalidCustomUrl] = React.useState(false);
+
+    const submit: React.FormEventHandler<HTMLFormElement> = (e) => {
+      e.preventDefault();
+      if (selectedRadio === 'custom') {
+        if (!customUrl) {
+          setInvalidCustomUrl(true);
+          return;
+        } else if (customUrl === currentUpstream) {
+          return handlePromise(Promise.resolve(), close);
+        }
+      } else if (!currentUpstream) {
+        return handlePromise(Promise.resolve(), close);
+      }
+      const value = selectedRadio === 'default' ? null : customUrl;
+      const patch = [{ op: 'add', path: '/spec/upstream', value }];
+      return handlePromise(k8sPatch(ClusterVersionModel, cv, patch), close);
+    };
+    const { t } = useTranslation();
+
+    return (
+      <form onSubmit={submit} name="form" className="modal-content ">
+        <ModalTitle>{t('public~Edit upstream configuration')}</ModalTitle>
+        <ModalBody>
+          <p>
+            {t(
+              'public~Select a configuration to receive updates. Updates can be configured to receive information from Red Hat or a custom update service.',
+            )}
+          </p>
+          <ExternalLink
+            href={'/'} //TODO
```

Comment:
Maybe https://docs.openshift.com/container-platform/4.8/updating/installing-update-service.html

Make sure to use `openshiftHelpBase`.